TiDB panics when using both the plugins 'Audit Log' and 'Extends Audit Log'  simultaneously

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1. deploy a enterprise tidb cluster with audit log plugin
2. enable audit log plugin and extends audit log
```sql
select audit_log_create_filter('all', '{}');
select audit_log_create_rule('%@%', 'all');
set global tidb_audit_enabled = 1;

insert into mysql.tidb_audit_table_access (user, db, tbl, access_type) values ('test', '.*', '.*', ''), ('^qa$', '.*', '.*', '');
flush tidb plugins audit;
create user test;
```
3. connet the tidb cluster with user `test` and execute sql
```sql
use test
```
<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)
both old and new audit log should record correctlly.
### 3. What did you see instead (Required)
tidb panic. If grant the privileges to the user, it will works.
```sql
mysql> use test;
ERROR 1105 (HY000): runtime error: invalid memory address or nil pointer dereference
```
tidb log 
```log
[2023/05/17 09:33:13.475 +00:00] [ERROR] [conn.go:1057] ["connection running loop panic"] [conn=4494121837139070629] [lastSQL="Use test"] [err="runtime error: invalid memory address or nil pointer dereference"] [stack="github.com/pingcap/tidb/server.(*clientConn).Run.func1
	/home/jenkins/agent/workspace/build-tidb-release-7.1/tidb/server/conn.go:1060
runtime.gopanic
	/usr/local/go/src/runtime/panic.go:884
runtime.panicmem
	/usr/local/go/src/runtime/panic.go:260
runtime.sigpanic
	/usr/local/go/src/runtime/signal_unix.go:837
github.com/pingcap/tidb/server.(*stmtEventInfo).ensureStmtContextOriginalSQL
	/home/jenkins/agent/workspace/build-tidb-release-7.1/tidb/server/extension.go:239
github.com/pingcap/tidb/server.(*stmtEventInfo).SQLDigest
	/home/jenkins/agent/workspace/build-tidb-release-7.1/tidb/server/extension.go:170
github.com/pingcap/tidb/extension/enterprise/audit.newStmtEventEntry.func1
	/home/jenkins/agent/workspace/build-tidb-release-7.1/tidb/extension/enterprise/audit/entry.go:388
github.com/pingcap/tidb/extension/enterprise/audit.(*LogEntry).Log
	/home/jenkins/agent/workspace/build-tidb-release-7.1/tidb/extension/enterprise/audit/entry.go:470
github.com/pingcap/tidb/extension/enterprise/audit.(*LogManager).GetSessionHandler.func2
	/home/jenkins/agent/workspace/build-tidb-release-7.1/tidb/extension/enterprise/audit/manager.go:659
github.com/pingcap/tidb/extension.(*SessionExtensions).OnStmtEvent
	/home/jenkins/agent/workspace/build-tidb-release-7.1/tidb/extension/session.go:145
github.com/pingcap/tidb/server.(*clientConn).onExtensionStmtEnd
	/home/jenkins/agent/workspace/build-tidb-release-7.1/tidb/server/extension.go:99
github.com/pingcap/tidb/server.(*clientConn).dispatch
	/home/jenkins/agent/workspace/build-tidb-release-7.1/tidb/server/conn.go:1343
github.com/pingcap/tidb/server.(*clientConn).Run
	/home/jenkins/agent/workspace/build-tidb-release-7.1/tidb/server/conn.go:1138
github.com/pingcap/tidb/server.(*Server).onConn
	/home/jenkins/agent/workspace/build-tidb-release-7.1/tidb/server/server.go:677"]
[2023/05/17 09:36:25.598 +00:00] [INFO] [domain.go:2634] ["refreshServerIDTTL succeed"] [serverID=2043690] ["lease id"=54ca882879068c09]
```
### 4. What is your TiDB version? (Required)
```sql
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| tidb_version()                                                                                                                                                                                                                                                                                |
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Release Version: v7.1.0
Edition: Enterprise
Git Commit Hash: 6d29491cec48777cf80f3fa24cf70b3ae9d1cd87
Git Branch: heads/refs/tags/v7.1.0
UTC Build Time: 2023-05-16 13:11:57
GoVersion: go1.20.3
Race Enabled: false
TiKV Min Version: 6.2.0-alpha
Check Table Before Drop: false
Store: tikv |
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```
<!-- Paste the output of SELECT tidb_version() -->

--- a/extension/event_listener_test.go
+++ b/extension/event_listener_test.go
@@ -120,6 +120,7 @@ type stmtEventCase struct {
 	parseError      bool
 	prepareNotFound bool
 	multiQueryCases []stmtEventCase
+	dispatchData    []byte
 }
 
 func TestExtensionStmtEvents(t *testing.T) {
@@ -324,6 +325,17 @@ func TestExtensionStmtEvents(t *testing.T) {
 			sql:        "create placement policy p1 followers=1",
 			redactText: "create placement policy `p1` followers = ?",
 		},
+		{
+			dispatchData: append([]byte{mysql.ComInitDB}, []byte("db1")...),
+			originalText: "use `db1`",
+			redactText:   "use `db1`",
+		},
+		{
+			dispatchData: append([]byte{mysql.ComInitDB}, []byte("noexistdb")...),
+			originalText: "use `noexistdb`",
+			redactText:   "use `noexistdb`",
+			err:          "[schema:1049]Unknown database 'noexistdb'",
+		},
 	}
 
 	for i, c := range cases {
@@ -342,6 +354,8 @@ func TestExtensionStmtEvents(t *testing.T) {
 			}
 		case c.binaryExecute != 0:
 			err = conn.Dispatch(context.Background(), getExecuteBytes(c.binaryExecute, false, true, c.executeParams...))
+		case c.dispatchData != nil:
+			err = conn.Dispatch(context.Background(), c.dispatchData)
 		}
 
 		if c.err != "" {

--- a/server/conn.go
+++ b/server/conn.go
@@ -1450,7 +1450,7 @@ func (cc *clientConn) useDB(ctx context.Context, db string) (node ast.StmtNode, 
 	}
 	_, err = cc.ctx.ExecuteStmt(ctx, stmts[0])
 	if err != nil {
-		return nil, err
+		return stmts[0], err
 	}
 	cc.dbname = db
 	return stmts[0], err

--- a/server/extension.go
+++ b/server/extension.go
@@ -235,7 +235,7 @@ func (e *stmtEventInfo) ensureStmtContextOriginalSQL() string {
 		e.sc.InitSQLDigest(planCache.NormalizedSQL, planCache.SQLDigest)
 	}
 
-	if e.sc.OriginalSQL == "" && e.executeStmtID == 0 {
+	if e.sc.OriginalSQL == "" && e.executeStmtID == 0 && e.stmtNode != nil {
 		e.sc.OriginalSQL = e.stmtNode.Text()
 	}
 

